File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *


input = stdin.readline

succ = "Enjoy other party\n"
fail = "Stay here\n"
ans = ""


n,m = map(int,input().split())
adj = []
prev = [[ i for _ in range(m)] for i in range(n)]



for _ in range(n):
    adj.append(list(map(int,input().split())))



for k in range(n):
    for i in range(n):
        for j in range(n):
            if adj[i][k] + adj[k][j] < adj[i][j]:
                prev[i][j] = prev[k][j]
                adj[i][j] = adj[i][k] + adj[k][j]

for _ in range(m):

    a,b,c = map(int,input().split())
    a -= 1
    b -= 1
    
    if adj[a][b] <= c:
        ans += succ
    else :
        ans += fail


print(ans[:-1])

File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.swift

import Foundation



let input = readLine()!.split{$0 == " "}.map{Int($0)!}

let (n,m) = (input[0],input[1])

var adj: [[Int]] = [[]]

for _ in 0..<n {
    adj.append([0] + readLine()!.split{$0 == " "}.map{Int($0)!})
}

let succ = "Enjoy other party"
let fail = "Stay here"

for k in 1...n {
    for i in 1...n {
        for j in 1...n {
            
            if adj[i][j] > adj[i][k] + adj[k][j] {
                adj[i][j] = adj[i][k] + adj[k][j]
            }
            
        }
    }
}

var result: [String] = []

for _ in 0..<m {
    let tmp = readLine()!.split{$0 == " "}.map{Int($0)!}
    
    let (a,b,c) = (tmp[0],tmp[1],tmp[2])
    
    if adj[a][b] <= c {
        result.append(succ)
    } else {
        result.append(fail)
    }
    
}


print(result.joined(separator: "\n"))
